src/typing/typingReducer.js

```diff
@@ -51,30 +92,10 @@ const eventTypingStop = (state, action) => {
     action.recipients,
     action.ownUserId,
   );
-  const previousTypingUsers = state[normalizedRecipients];
+  const senderId = action.sender.user_id;
+  const time = action.time;
 
-  if (!previousTypingUsers) {
-    return state;
-  }
-
-  const newTypingUsers = state[normalizedRecipients].userIds.filter(
-    userId => userId !== action.sender.user_id,
-  );
-
-  if (newTypingUsers.length > 0) {
-    return {
-      ...state,
-      [normalizedRecipients]: {
-        time: action.time,
-        userIds: newTypingUsers,
-      },
-    };
-  }
```

Comment:
This isn't a pure refactor of the `eventTypingStop` handler, but it
    makes it better by fixing a small bug. Before this change, if I'm in
    a group PM with many people, and several are typing at once, an
    EVENT_TYPING_STOP for a single person will clear the typing state of
    *everyone* who was typing, not just the person who stopped typing.

Huh, really? This code looks to me like it correctly handled that case. What am I missing?